README.md

```diff
@@ -1,17 +1,151 @@
+├── contracts            # smart contracts solidity files
+├── scripts               # deployment scripts (deploy.js) and other tasks
+├── test                  # test scripts
+├── .env.example          # environment variables template
+├── .gitignore
+├── hardhat-config.js     # hardhat configuration
+├── package.json          # project details and dependencies
+├── README.md
+
+```
+
+## Instructions and hardhat commands
+
+- Install the autofarm project dependencies. \
+  This will install the packages mentioned inside the `package.json` file.
+
+  ```shell
+  npm i
+  ```
+
+- Create the `.env` file following the `.env.example` file and fill in the contents. The values for the `ALCHEMY_API_KEY` can be obtained from the [Alchemy](https://www.alchemy.com/) website. If you don't have an account, you can create one for free.
+
+  ```shell
+  cp .env.example .env
+  ```
+
+  Else, you can comment out the places in the `hardhat.config.ts` file where the `ALCHEMY_API_KEY` is used, to be able to compile the smart contracts.
+
+- Compile the smart contracts. \
+  This will compile the smart contract files using the specified soilidity compiler in `hardhat.config.ts`.
+
+  ```shell
+  npx hardhat compile
+  ```
+
+- Instantiate the hardhat local node.
+
+  ```shell
+  npx hardhat node
+  ```
+
+- Run the autofarm project tests using the local node.
+
+  ```shell
+  npx hardhat test --network localhost
+  ```
+
+- Generate the code coverage report. \
+  After generating the report, you can open the `coverage/index.html` file to see the results.
+
+  ```shell
+  npx hardhat coverage
+  ```
+
+- Deployment of the smart contracts. \
+  There are two ways of deploying the smart contracts:
+
+  - Using private keys: \
+    The private key needs to be pasted in the .env file and the `accounts` property in the `networks` properties in the `hardhat.config.ts` file needs to be uncommented for the network on which the smart contracts need to deployed, as follows.
+
+    ```shell
+    goerli: {
+      url: `https://eth-goerli.g.alchemy.com/v2/${ALCHEMY_API_KEY}`,
+      accounts: [`0x${PRIVATE_KEY}`],
+      chainId: 5,
+    },
+    ```
+
+  - Without using private keys: \
+    This lets you deploy the contracts without the need of pasting the private key anywhere in the project and is the preferred way of deployment.
+
+    - Install truffle globally
+
+      ```shell
+      npm install -g truffle
+      ```
+
+    - Run the truffle dashboard
+
+      ```shell
+      truffle dashboard
+      ```
+
+    - The browser will open up and then you have to connect with the MetaMask extension. Select the preferred network and the account to deploy the smart contract.
+
+- Deploy the autofarm project smart contracts using either of the two ways mentioned above.
+
+  ```shell
+  npx hardhat run .\scripts\autofarm.deploy --network <NETWORK>  param1 param2
+  ```
+
+  ```shell
+  npx hardhat run .\scripts\stratx2.deploy --network <NETWORK> param1 param2
+  ```
```

Comment:
what are these `param1` and `param2`?

Reply:
arguments to pass for running the function

Reply:
mention in detail in the readme what type of argument it can be, with examples.